include/graaflib/algorithm/cycle_detection.tpp

```diff
@@ -0,0 +1,102 @@
+#pragma once
+
+#include <graaflib/types.h>
+
+#include <algorithm>
+#include <unordered_map>
+
+namespace graaf::algorithm {
+
+enum vertex_color { UNVISITED, VISITED, NO_CYCLE };
+namespace detail {
+
```

Comment:
I would still make this an `enum class` rather than an `enum`. Then the values do not pollute the namespace it is in. I would then also move it inside the detail namespace to keep the scope as small as possible.
```suggestion

namespace detail {

enum class vertex_color { UNVISITED, VISITED, NO_CYCLE };

```